OCPBUGS#5244: When an admin user associates a floating IP to the reservation port, delete operation hangs

xenolinux opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.12+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: https://issues.redhat.com/browse/OCPBUGS-5244
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://58176--docspreview.netlify.app/openshift-enterprise/latest/networking/openshift_sdn/assigning-egress-ips.html#nw-egress-ips-public-cloud-platform-considerations_egress-ips
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

rlobillo commented:
@andreaskaris If I am understanding correct, the scenario you are mentioning is only applicable for 4.12. Thanks to below 4.13 epic: 
https://issues.redhat.com/browse/OSASINFRA-3007

The port is not recreated on an failover scenario. Do you think there are other scenarios that can lead to the problem described in https://issues.redhat.com/browse/OCPBUGS-5244 ?

Thanks.

dulek commented:
@rlobillo: This is a different issue from the one you think about. This one is when it's admin that created the FIP and assigned it to the port. Then the cloud-network-config-controller wouldn't be able to detach it from the port. It still shouldn't be a 500 error from Neutron, but in general it's not something we support, operations have to be done from the same tenant.

dulek commented:
@xenolinux: Current version has little sense to me, @andreaskaris explained it better a few comments above.

Also please pay attention that you've edited this *inside* the `ifdef::openshift-sdn[]` block. The problem you're trying to describe happens with both openshift-sdn and ovn-kubernetes. The problem with lost floating IP assignment on failover currently only happens with openshift-sdn, that's why we have the ifdef block.

xenolinux commented:
Hi @andreaskaris I updated the PR. Moved the text out of the `ifdef::openshift-sdn[]` block since this scenario is valid for both openshift-sdn and ovn-kubernetes. Please let me know if any other change is required.

dulek commented on modules/nw-egress-ips-about.adoc:
```diff
@@ -88,6 +88,11 @@ The {rh-openstack} egress IP address feature creates a Neutron reservation port
 
+[NOTE]
+====
+When a cluster administrator assigns a floating IP to the reservation port, {product-title} cannot delete the reservation port. The `CloudPrivateIPConfig` object cannot perform delete and move operations until the floating IP is unassigned from the reservation port.
```
Can we make it more specific that this is about OpenStack cluster administrator, not OpenShift? Right now this is ambiguous.

I think we can use `{rh-openstack}` to achieve that.

xenolinux replied:
@dulek Addressed.

xenolinux replied:
Does the note look good now?

dulek replied:
Looks good from my perspective.

dulek reviewed (changes requested):
So I'm not 100% sure move EgressIP operation is affected, but it's totally fine to play it safe and add it. In general the bottom line of this note is - don't assign FIPs using admin accounts.

Tiny remark inline. Besides that this is good to go.

andreaskaris commented:
/lgtm

rlobillo commented:
/lgtm

xenolinux commented:
/label peer-review-needed

abhatt-rh reviewed (commented):
LGTM, except a minor comment!

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

xenolinux commented:
/label merge-review-needed

snarayan-redhat commented:
/cherrypick enterprise-4.12

snarayan-redhat commented:
/cherrypick enterprise-4.13
